src/main/java/dev/openfeature/sdk/ProviderEventDetails.java

```diff
@@ -0,0 +1,18 @@
+package dev.openfeature.sdk;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import lombok.Data;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * Interface for attaching event handlers.
+ */
+@Data @SuperBuilder(toBuilder = true)
+public class ProviderEventDetails {
+    @Nullable private List<String> flagsChanged;
+    @Nullable private String message;
+    @Nullable private FlagMetadata eventMetadata; // TODO: this should use a different type.
```

Comment:
@Kavindu-Dodan this field essentially has to be the same thing as your recently implemented `FlagMetadata`... however, the name `FlagMetadata` doesn't make sense.

Reply:
Will rebase on https://github.com/open-feature/java-sdk/pull/478